### PR TITLE
String Builder

### PR DESCRIPTION
**Pull request by kodomazer**
Changed string concatenation into String Builders - #113 

I've been on hiatus since I'm busy with school, and I'm only set to get busier.  Crunch time is coming and I've returned here to relax a bit.  I see you guys have been busy without me.

Hopefully this change doesn't break anything... not sure how it would.

### CODE REVIEW

**Comment by PRabahy** [comment]
I don't really have an objection to this, but both loops will only iterate a very small number of times, so it doesn't impact performance either way.

File: src/evolv/io/Creature.java [hunk 1]
```diff
@@ -364,25 +364,25 @@ public void reproduce(double babySize, double timeStep) {
 	}
 
 	public String stitchName(String[] s) {
-		String result = "";
+		StringBuilder build = new StringBuilder();
```
Comment (PRabahy) [inline src/evolv/io/Creature.java]: "build" is a verb. Usually I like using nouns for variables and verbs for methods. I would prefer this be named "builder".

**Comment by kodomazer** [comment]
Updated build to builder.

I mainly wanted to find something small to do so I wouldn't have to think too much about it.

Might possibly want to make a project board for tasks we need to complete for cleanup and the such.

File: src/evolv/io/Creature.java [hunk 2]
```diff
@@ -364,25 +364,25 @@ public void reproduce(double babySize, double timeStep) {
...
 
 	public String stitchName(String[] s) {
-		String result = "";
+		StringBuilder builder = new StringBuilder();
 		for (int i = 0; i < s.length; i++) {
 			float portion = ((float) s[i].length()) / s.length;
 			int start = Math.min(Math.max(Math.round(portion * i), 0), s[i].length());
 			int end = Math.min(Math.max(Math.round(portion * (i + 1)), 0), s[i].length());
-			result = result + s[i].substring(start, end);
+			builder.append(s[i].substring(start, end));
```
Comment (NateUniAccount) [inline src/evolv/io/Creature.java]: The substring() method creates a new String, there's a StringBuilder method which takes a String and a start and end index and appends the characters in between which will be more efficient

**Comment by kodomazer** [comment]
Thank you for bearing with me.

It feels nice working with people who are more competent than me.

**Comment by NateUniAccount** [comment]
No worries! haha flattery will get you everywhere :D

**Pull request merged**